src/lazy/binary/raw/v1_1/value.rs

```diff
@@ -194,7 +194,13 @@ impl<'top> LazyRawValue<'top, BinaryEncoding_1_1> for &'top LazyRawBinaryValue_1
             return Ok(ValueRef::Int(int));
         }
         if self.is_null() {
-            return Ok(ValueRef::Null(self.ion_type()));
+            let ion_type = if self.encoded_value.header.ion_type_code == OpcodeType::TypedNull {
+                let body = self.value_body();
+                ION_1_1_TYPED_NULL_TYPES[body[0] as usize]
+            } else {
+                IonType::Null
+            };
+            return Ok(ValueRef::Null(ion_type));
```

Comment:
Why was this needed? `ion_type()` should return the correctly interpreted type from the opcode and optional typed null byte. (`return Ok(ValueRef::Null(self.ion_type()))` also appears below on line 227).

Reply:
When parsing the Opcode we don't have the byte containing the true ion type ([here](https://github.com/amazon-ion/ion-rust/blob/main/src/lazy/binary/raw/v1_1/type_descriptor.rs#L79)), so the actual ion type was originally deferred until the value is read ([here](https://github.com/amazon-ion/ion-rust/blob/main/src/lazy/binary/raw/v1_1/value.rs#L147)). Right now `ion_type()` just returns the type from the opcode, but you're right, it's a better place to be handling this logic.